Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -1,13 +1,34 @@
 class JavaRequirement < Requirement
-  cask "java"
+  def initialize(tags)
+    @version = tags.shift if /(\d+\.)+\d/ =~ tags.first
+    @cask = cask
+    super
+  end
 
   env do
     env_java_common
     env_oracle_jdk || env_apple
   end
 
+  # A strict Java 8 requirement (1.8) should prompt the user to install java8
```

Comment:
This describes what it does but not why it's doing what it's doing. The comment is relatively obvious from reading the code but it would be good to explain the whole "Java is usually backwards compatible but 9.0 is causing everything to explode due to breaking changes and changing version number format" stuff here instead.